search/includes/classes/class-versioning.php

```diff
@@ -745,4 +755,128 @@ public function action__ep_delete_indexable( $object_id, $indexable_slug ) {
 		// Clear the flag to return to normal
 		$this->is_doing_object_delete = false;
 	}
+
+	/**
+	 * Check if the versions are persisted correctly. If not recreate them.
+	 */
+	public function maybe_self_heal() {
+		if ( $this->is_versioning_valid() ) {
+			return;
+		}
+
+		// $indicies = $this->get_all_accesible_indicies();
+		// $versioning = $this->reconstruct_versioning( $indicies );
+
+		// update_option( self::INDEX_VERSIONS_OPTION, $versioning );
+	}
+
+	public function is_versioning_valid() {
+		$all_versions = get_option( self::INDEX_VERSIONS_OPTION, array() );
+
+		if ( ! is_array( $all_versions ) || count( $all_versions ) == 0 ) {
+			return false;
+		}
+
+		$valid_slugs = \ElasticPress\Indexables::factory()->get_all( null, true );
+
+
+		foreach ( $all_versions as $slug => $versions ) {
+			if ( ! in_array( $slug, $valid_slugs ) ) {
+				return false;
```

Comment:
I think we can omit this check because this will end up triggering a rebuild when features are disabled, which is probably a worse user experience (since the active index could end up being different after the rebuild, if the previously active one wasn't the oldest) than letting the data stay there and possibly be stale if that feature was re-activated. Thoughts?

Reply:
Sounds good, I dont see any harm in keeping unmatched ones there.